### PR TITLE
Post-novo express cleanup

### PR DESCRIPTION
**Pull request by damassi**
This cleans up a bunch of WIP/Todo-ish stuff in our express folder that crept in during novo work, such as 
- consolidating all middleware into one location, versus three
- relocating files and context where appropriate (dev stuff to dev, middleware stuff to middleware, etc)
- flattening out and simplifying some of the execution by removing unnecessary callback code, wrapping functions, etc
- renaming novo routes to regular routes

Also removed / collapsed a couple extraneous files for clarity. The root now consists of 
```
- src/config
- src/dev
- src/index
- src/middleware
```

#### This is the last PR in a family of related work:
- Webpack Spring Cleaning #7225
- Rename clientConfig to legacyConfig and novoConfig to clientConfig #7231
- Move novo files to v2 #7233
- Add React fast-refresh for hot reloading #7235

#### Browse files:
- [Before](https://github.com/artsy/force/tree/832f84ec011a45d8baef158a82110979279b031e/src)
- [After](https://github.com/artsy/force/tree/6e51f77d79a64f8c87c5f568c4026ff3a31c8640/src)

#### Review App:
- https://cleanup.artsy.net/

#### Whats left:
- Remove references to `novo-assets` output, if possible.  (Though I think it's required as we have two parallel webpack streams now.)

### CODE REVIEW

File: src/lib/cache.js [hunk 1]
```diff
@@ -128,6 +129,4 @@ class Cache {
   }
 }
 
-const cache = new Cache()
-
-module.exports = cache
+export const cache = new Cache()
```
Comment (damassi) [inline src/lib/cache.js:132]: Not sure why, but babel did not like this being exported with `module.exports`, leading to some of the churn around cache imports throughout

File: cypress/integration/artist.spec.js [hunk 2]
```diff
@@ -20,6 +20,5 @@ describe("/artist/:id", () => {
     cy.get("h1").should("contain", "Pablo Picasso")
     cy.get("h2").should("contain", "Spanish, 1881–1973")
     cy.get("h2").should("contain", "Notable Works")
-    cy.get("h2").should("contain", "Iconic Collections")
```
Comment (damassi) [inline cypress/integration/artist.spec.js:23]: Rail is lazy loaded via scroll interception on the client, and far down the page

Comment (damassi) [inline cypress/integration/artist.spec.js:23]: This is a lazy loaded rail outside of the initial view, triggered on scroll: 
Screen Shot 2021-03-28 at 1 33 01 PM

Def not critical as we're only making sure the page loads

File: src/lib/middleware/errorHandler.ts [hunk 3]
```diff
@@ -47,3 +48,23 @@ export function errorHandlerMiddleware(
 
   res.status(code).render(file, { code, detail, message })
 }
+
+export function setupErrorHandling(app) {
```
Comment (icirellik) [inline src/lib/middleware/errorHandler.ts:52]: I know there was a comment in the original file to look into moving this, however, this feels far less intuitive than leaving it as it was.

File: src/lib/middleware/errorHandler.ts [hunk 4]
```diff
@@ -47,3 +48,23 @@ export function errorHandlerMiddleware(
...
+    err.message = "Not Found"
+    next(err)
+  })
+
+  if (SENTRY_PRIVATE_DSN) {
+    // Old Sentry SDK.
+    app.use(RavenServer.errorHandler())
+
+    RavenServer.config(SENTRY_PRIVATE_DSN).install()
+    app.use(RavenServer.requestHandler())
```
Comment (icirellik) [inline src/lib/middleware/errorHandler.ts:66]: I have a feeling that this needs to stay as one of the first middleware added to properly hook into force.

File: webpack/index.js [hunk 5]
```diff
@@ -116,22 +116,23 @@ function generateEnvBasedConfig() {
...
+  } else if (config === "server.dev") {
+    return serverConfig
+  } else if (config === "server.prod") {
+    return serverConfig
   }
 }
+
+// Currently only used for prod
+const webpackConfig = generateEnvBasedConfig()
+export default webpackConfig
```
Comment (icirellik) [inline webpack/index.js:138]: 💖 I love that this solves the `AUTO_CONFIGURE` issue!

**Review by dzucconi — APPROVED** [review]

**Review by icirellik — COMMENTED** [review]
There are a couple of comments that I would love to see addressed however this PR has trampled the novo cleanup I was working on Thursday and as I have additional, now blocked, changes that finish the novo cleanup...yolo merge 😄 ?

In the future, we should assign tickets to a single engineer as there has been a significant amount of wasted effort on this task.

Additionally, would you use `git mv` or make a separate commit when remaining files with large changes to maintain the link between them. e.g. diffing `index.dev.js` and `dev.js` is really difficult.

File: webpack/index.js [hunk 6]
```diff
@@ -13,7 +13,7 @@ import { clientDevelopmentConfig } from "./envs/clientDevelopmentConfig"
 import { clientProductionConfig } from "./envs/clientProductionConfig"
 import { serverConfig } from "./envs/serverConfig"
 
-const getServerConfig = () => {
+function getServerConfig() {
```
[the remark this answers is not in the document]
Comment (damassi) [reply webpack/index.js:16]: This is so we get better stack traces should things go wrong, as well as makes it consistent with rest of file (which uses `function` keyword)

File: src/lib/cache.js [hunk 7]
```diff
@@ -20,6 +20,7 @@ class Cache {
       return callback()
     }
     if (OPENREDIS_URL) {
+      const redis = require("redis")
```
[the remark this answers is not in the document]
Comment (damassi) [reply src/lib/cache.js:23]: Webpack was excluding it via the `externals` but for some reason an error was being thrown on the client, I think due to react hot loader. This is a fine move, "if we've got a redis url, load redis". We don't touch this file often.

File: src/middleware.ts [hunk 8]
```diff
@@ -41,6 +64,94 @@ const {
...
+
+  // Redirect requests before they even have to deal with Force routing
+  app.use(downcaseMiddleware)
+  app.use(hardcodedRedirectsMiddleware)
+  app.use(localsMiddleware)
+  app.use(backboneErrorHandlerMiddleware)
+  app.use(sameOriginMiddleware)
+  app.use(escapedFragmentMiddleware)
+  app.use(unsupportedBrowserMiddleware)
+  app.use(pageCacheMiddleware)
```
[the remark this answers is not in the document]
Comment (damassi) [reply src/middleware.ts:111]: Everything was spread all over in a confusing way, and in the end, the new app is simply an express route (`/`) mounted with an express sub app (v2 folder). So its all the same. We can add a comment in a follow-up pr.

File: src/lib/middleware/errorHandler.ts (hunk 3 above)
Comment (damassi) [reply src/lib/middleware/errorHandler.ts:52]: Why? There's a function called `setupErroHandling` right where this block of code would be, only now its been moved to the `errorHandler` file, which actually handles errors. So its logical and hides away unnecesary implementation details by co-locating it with its sibling.

File: src/lib/middleware/errorHandler.ts (hunk 4 above)
Comment (damassi) [reply src/lib/middleware/errorHandler.ts:66]: Let me look at the docs, I remember a while back it said it had to go last, but let me see.

Comment (damassi) [reply src/lib/middleware/errorHandler.ts:66]: Ah, yup: https://docs.sentry.io/clients/node/integrations/#express

Will fix

File: src/lib/middleware/errorHandler.ts (hunk 3 above)
Comment (damassi) [reply src/lib/middleware/errorHandler.ts:52]: With that raven line needing to be first, will move this stuff back in

File: src/lib/cache.js (hunk 1 above)
Comment (damassi) [reply src/lib/cache.js:132]: This file crosses the client/server boundary in subtle ways, so possibly related to that. In any case, better than `module.exports`, just kinda weird and annoying.

**Comment by damassi** [comment]
Comments have been addressed and will remember `git mv` in the future. (For now, it just consolidates all of our dev related aspects into one location, versus being spread all around.)

**Comment by damassi** [comment]
Going to merge to get this to staging. If there's any further needs we can address it in a follow-up PR.

**Pull request merged**